### PR TITLE
Add jinja2 + parameterization tutorial.

### LINKED ISSUE #1991: rose advanced tutorial: parametrization + jinja2
Following the recent addition of task parametrization to cylc and our desire to get people adopting it, it would be good to produce a rose advanced tutorial for this (perhaps as a follow on from the jinja2 one?). We may also wish to update the main documentation examples which currently use jinja2, to use parametrization instead and highlight using jinja2 for more advanced cases.

### PR DESCRIPTION
**Pull request by arjclark**
Close #1991

### CODE REVIEW

File: doc/rose-rug-advanced-tutorials-parameterization-and-jinja2.html [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+              <p>By combining Jinja2 with task parameterization you can make
+              use of further flexibility in generating your
+              <var>suite.rc</var>.</p>
+
+            </div>
+
+            <div class="slide">
+              <h2 id="purpose">Purpose</h2>
+
+              <p>The purpose of this tutorial is to demonstrate some the
```
Comment (matthewrmshin) [inline doc/rose-rug-advanced-tutorials-parameterization-and-jinja2.html]: some *of* the

**Comment by arjclark** [comment]
@matthewrmshin - missing "of" added

File: doc/rose-rug-advanced-tutorials-parameterization-and-jinja2.html [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+              guide.</p>
+
+              <p>To start with, we will extend the suite so that a user can
+              pass through a set of worlds to say hello to via an entry in the
+              <var>rose-suite.conf</var> file.</p>
+
+            </div>
+
+            <div class="slide">
+              <h3 id="example-2">Example (2)</h3>
```
Comment (oliver-sanders) [inline doc/rose-rug-advanced-tutorials-parameterization-and-jinja2.html]: I think these headings should only appear in sideshow mode, to do so add the class `slides-only` and remove the id.

**Comment by arjclark** [comment]
@oliver-sanders - changes made.

**Pull request merged**